pkgs/tools/misc/as-tree/default.nix

```diff
@@ -0,0 +1,25 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "as-tree";
+  version = "0.12.0";
+
+  src = fetchFromGitHub {
+    owner = "jez";
+    repo = pname;
+    rev = version;
+    sha256 = "0c0g32pkyhyvqpgvzlw9244c80npq6s8mxy3may7q4qyd7hi3dz5";
+  };
+
+  cargoSha256 = "0g3x76l251g82jrd194whdsfxzlg45bhr1y6bjjfc79viz9mrr87";
+  cargoPatches = [ ./cargo-lock.patch ];
```

Comment:
As far as I understand the purpose of this patch is to update the erroneous version of as-tree in Cargo.lock but it also updates the (minor) versions of other crates.
Might I suggest updating the `rev` in `fetchFromGitHub` to use the [upstream fix](https://github.com/jez/as-tree/commit/c82bab213ad11fa52a7827146b506f44aeef4c25) instead? I (personally) think it's cleaner.

Reply:
Possibly: I saw that commit also, but I wanted to stick to the tagged release commit. Ideally, I suppose, upstream would have a 0.12.1 release to fix this. I'm new to packaging Rust applications for Nix. Perhaps the ideal middle ground is to change the patch so it only fixes the wrong version of as-tree, and leaves the other crates as-is?

Reply:
Fine with me. Maybe also leave a small note above cargoPatches to indicate why the patch exists and to remove it when a new version releases.

Reply:
I have added an explanatory comment and reduced the patch to only update the version of as-tree itself